cpp/src/arrow/adapters/orc/CMakeLists.txt

```diff
@@ -25,7 +25,12 @@ install(FILES adapter.h DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/arrow/adapters/
 # pkg-config support
 arrow_add_pkg_config("arrow-orc")
 
-set(ORC_MIN_TEST_LIBS GTest::Main GTest::GTest)
+set(ORC_MIN_TEST_LIBS
+    GTest::Main
+    GTest::GTest
+    Snappy::snappy
+    LZ4::lz4
+    ZLIB::ZLIB)
```

Comment:
Do we need `ARROW_WITH_XXX` check?

For example:

```cmake
if(ARROW_WITH_LZ4)
  list(APPEND ORC_MIN_TEST_LIBS LZ4::lz4)
endif()
```

Reply:
Hmm, I don't think so? LZ4 is not optional when building ORC, AFAICT.

Reply:
And we have this in `cpp/CMakeLists.txt`:
```cmake
if(ARROW_ORC)
  set(ARROW_WITH_LZ4 ON)
  set(ARROW_WITH_SNAPPY ON)
  set(ARROW_WITH_ZLIB ON)
endif()
```

Reply:
Thanks. I understand.